include/clad/Differentiator/Tape.h

```diff
@@ -26,68 +32,88 @@ namespace clad {
     using iterator = pointer;
     using const_iterator = const_pointer;
    
+    
+    #ifdef __CUDACC__
+      void* operator new( size_t size, unsigned int extra) noexcept {}
+    #endif
+
+    /// Move values from old to new storage
+    CUDA_HOST_DEVICE T* MoveValues(std::size_t _capacity) {
+      #ifdef __CUDACC__
+        // Allocate raw storage (without calling constructors of T) of new capacity.
+        T* new_data = static_cast<T*>(::operator new(_capacity * sizeof(T)));
+        assert(new_data);
+        // Move values from old storage to the new storage. Should call move
+        // constructors on non-trivial types, otherwise is expected to use
+        // memcpy/memmove.
+        for (auto i=begin(); i != end(); ++i, ++new_data) {
```

Comment:
```suggestion
        for (auto i=begin(), e = end(); i != e; ++i, ++new_data) {
```
This way we tell the reader that we do not modify the collection in the loop.